File: php/testcode/xss_015.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_xss_stored_db
function xss_stored_db(BenchmarkRequest $req): BenchmarkResponse {
    $id = $req->param('id');
    $pdo = getDbConnection();

    $stmt = $pdo->prepare('SELECT bio FROM users WHERE id = ?');
    $stmt->execute([$id]);
    $row = $stmt->fetch();
    $bio = $row['bio'] ?? '';

    $html = '<div class="bio">' . $bio . '</div>'; // vuln-code-snippet vuln-line php_xss_stored_db

    return BenchmarkResponse::html($html);
}
// vuln-code-snippet end php_xss_stored_db

File: ruby/testcode/xss_015.rb
require_relative 'shared'

# vuln-code-snippet start ruby_xss_href_javascript
def xss_href_javascript(req)
  url = req.param('url')
  link_text = escape_html(req.param('text', default: 'Click here'))
  anchor = "<a href=\"#{url}\">#{link_text}</a>" # vuln-code-snippet vuln-line ruby_xss_href_javascript
  BenchmarkResponse.html("<nav>#{anchor}</nav>")
end
# vuln-code-snippet end ruby_xss_href_javascript
